### PR TITLE
Lucuma ODB

### PR DESCRIPTION
**Pull request by rpiaggio**
There is still work to be done but I think this is in a mergeable state, so as not too diverge further.

### CODE REVIEW

File: common/src/main/scala/explore/GraphQLSchemas.scala [hunk 1]
```diff
@@ -3,13 +3,39 @@
...
+      type Long          = scala.Long
+      type ObservationId = String
+      type ProgramId     = String
+      type TargetId      = Target.Id
+
+      implicit val encoderTargetId: Encoder[Target.Id] =
+        Encoder.encodeString.contramap[Target.Id](id => s"t-${id.toInt}")
+      implicit val decoderTargetId: Decoder[Target.Id] = Decoder.decodeString.emapTry { str =>
+        Try(Target.Id(str.drop(2).toInt))
+      }
```
Comment (rpiaggio) [inline common/src/main/scala/explore/GraphQLSchemas.scala:33]: This is temporary, as well as all Id logic. We will use `Gid` when we define its instances in `lucuma-core`.

File: explore/src/main/scala/explore/Routing.scala [hunk 2]
```diff
@@ -24,14 +25,14 @@ case object LabelsElement extends ElementItem
...
 
-  private val targetPageIso: Iso[SiderealTarget.Id, TargetPage] =
-    Iso[SiderealTarget.Id, TargetPage](TargetPage.apply)(_.targetId)
+  private val targetPageIso: Iso[Int, TargetPage] =
+    Iso[Int, TargetPage](i => TargetPage(Target.Id(i)))(_.targetId.toInt)
 
-  private val targetObsPageIso: Iso[ExploreObservation.Id, TargetsObsPage] =
-    Iso[ExploreObservation.Id, TargetsObsPage](TargetsObsPage.apply)(_.obsId)
+  private val targetObsPageIso: Iso[Int, TargetsObsPage] =
+    Iso[Int, TargetsObsPage](i => TargetsObsPage(s"o-$i"))(_.obsId.drop(2).toInt)
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/Routing.scala:35]: All this with change to `Gid`s.

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 3]
```diff
@@ -109,9 +108,9 @@ object ObsTabContents {
                     ^.cls := "tile",
                     Tile("Target Position", movable = true)(
                       <.span(
-                        obsSummaryOpt.whenDefined(obs =>
-                          TargetEditor(obs.target.id).withKey(obs.target.id.toString)
-                        )
+                        // obsSummaryOpt.whenDefined(obs =>
+                        //   TargetEditor(obs.target.id).withKey(obs.target.id.toString)
+                        // )
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:113]: Target editing in observation view is disabled until we deal with asterisms.

File: explore/src/main/scala/explore/tabs/ObsTabContents.scala [hunk 4]
```diff
@@ -120,11 +119,11 @@ object ObsTabContents {
...
-                        obsSummaryOpt.whenDefined(obs =>
-                          ConstraintsSubscription(obs.constraints.id) { constraints =>
-                            ConstraintsPanel(obs.constraints.id, constraints)
-                          }.withKey(obs.constraints.id.toString)
-                        )
+                        // obsSummaryOpt.whenDefined(obs =>
+                        //   ConstraintsSubscription(obs.constraints.id) { constraints =>
+                        //     ConstraintsPanel(obs.constraints.id, constraints)
+                        //   }.withKey(obs.constraints.id.toString)
+                        // )
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/tabs/ObsTabContents.scala:126]: We don't have constraints yet in ODB.

File: observationtree/src/main/scala/explore/observationtree/TargetObsList.scala [hunk 5]
```diff
@@ -54,65 +50,68 @@ object TargetObsList {
...
-              .composeOptionLens(ObsSummary.target)
-              .set(targetOpt.map(TargetSummary.fromTarget))
-          ) >>
-          // 2) Send mutation
-          mutateObs(obsId, ObservationsSetInput(target_id = targetOpt.map(_.id)))
+              .composeOptionLens(ObsIdNameTarget.target)
+              .set(targetOpt) //.map(TargetSummary.fromTarget))
+          ) //>>
+    // 2) Send mutation
+    // updateObs(EditObservationInput(obsId, targets = targetOpt.map(_.id)).some)
```
Comment (rpiaggio) [inline observationtree/src/main/scala/explore/observationtree/TargetObsList.scala:92]: We have to deal with asterisms in drag'n'drop now.

File: project/Settings.scala [hunk 6]
```diff
@@ -8,6 +8,7 @@ object Settings {
   object LibraryVersions {
     val cats              = "2.2.0"
     val catsEffect        = "2.2.0"
+    val chimney           = "0.5.3"
```
Comment (rpiaggio) [inline project/Settings.scala:11]: I'm not entirely sure we'll end up using chimney. I'm experimenting.

File: common/src/main/scala/explore/GraphQLSchemas.scala (hunk 1 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/GraphQLSchemas.scala]: Disregard comment above, already migrated to `Gid`.

File: explore/src/main/scala/explore/Routing.scala (hunk 2 above)
Comment (rpiaggio) [reply explore/src/main/scala/explore/Routing.scala]: Disregard comment above, already migrated to `Gid`.

**Review by cquiroz — APPROVED** [review]
I declare PR bankruptcy. LGTM

**Comment by toddburnside** [comment]
> I declare PR bankruptcy. LGTM

There is a lot of pretty green and red. A _**lot**_ of pretty green and red.

**Pull request merged**